dataset_loaders/videos/davis2017.py

```diff
@@ -0,0 +1,269 @@
+from __future__ import division
+import os
+import time
+
+import numpy as np
+import cv2
+from PIL import Image
+
+from dataset_loaders.parallel_loader import ThreadedDataset
+from dataset_loaders.utils_parallel_loader import natural_keys
+
+floatX = 'float32'
+
+
+class Davis2017Dataset(ThreadedDataset):
+    name = 'davis2017'
+    # Non void classes?
+    non_void_nclasses = None
+    _void_labels = []
+
+    # NOTE: we only load the 480p
+    # 1080p images are either (1920, 1080) or (1600, 900)
+    data_shape = (480, 854, 3)
+    # _cmap = {
+    #     0: (255, 255, 255),        # background
+    #     1: (0, 0, 0)}              # foreground
+    # _mask_labels = {0: 'background', 1: 'foreground'}
+
+    _unique_rgbs = None
+    _filenames = None
+    _image_sets_path = None
+    _prefix_list = None
+
+    @property
+    def prefix_list(self):
+        if self._prefix_list is None:
+            # Create a list of prefix out of the number of requested videos
+            this_set = 'val' if self.which_set == 'valid' else self.which_set
+            with open(os.path.join(self._image_sets_path,
+                                   this_set+'.txt')) as f:
+                content = f.readlines()
+                f.close()
+            self._prefix_list = [prefix.strip() for prefix in content]
+
+        return self._prefix_list
+
+    @property
+    def unique_rgbs(self):
```

Comment:
Actually it seems to be no time expensive but saving the RGBs in a file is a good idea. Is the numpy file supposed to be saved in the same folder of the dataset (shared path)?

Reply:
Good idea, it seems reasonable to put it with the dataset.

I suggest to follow the usual shared/local path philosophy, i.e., 1) check in the `__init__` if the file is in the *shared_path* and create it otherwise, 2) call `super()` as usual to copy the shared path in the local path 3) after the call, initialize `self.unique_rgbs` with the content of the file in the *local* path.

At your discretion, you could also skip 3) and read the file directly every time you access self.unique_rgbs to avoid storing it in memory, but it shouldn't be too big I think.